mathesar_ui/src/components/schema-row/SchemaRow.svelte

```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+  import {
+    faLock,
+    faProjectDiagram,
+  } from '@fortawesome/free-solid-svg-icons';
+  import { Icon } from '@mathesar-components';
+
+  // Props
+  export let schemaName = '';
+  export let tableCount = 0;
+  export let isDefault = true;
+  export let isLocked = true;
+
+  // Additional classes
+  let classes = '';
+  export { classes as class };
+
+  // Inline styles
+  export let style = '';
+</script>
+
+<div class={['schema-row', classes].join(' ')} {style}>
+  <div class="details">
+    <div class="title">
+      <Icon data={faProjectDiagram}/>
+      {schemaName}
+      {#if isLocked}
+        <Icon class="lock" data={faLock}/>
+      {/if}
+    </div>
+    <div class="info">
+      {#if isDefault}
+        <strong>Default</strong>
+        &middot;
+      {/if}
+      {tableCount} Tables
+    </div>
+  </div>
+  <div class="controls">
+    <slot/>
```

Comment:
For the controls, instead of a slot, we could keep the controls here, and dispatch events to the parent.

It would be best not to mix templates between parent and child components using slots, unless it is essential/reusable (like the modal).

Also, we could pass the entire schema object to this component, as a prop, instead of spreading it out. This would allow future changes with respect to the SchemaRow, stay only within this component.

Reply:
Moved the controls to the `SchemaRow` component in b84e1ba. The reason I went with `<slot/>` was precisely that, I expected `SchemaRow` to be a reusable component. With that changed in 775f800, this was no longer needed.